go/tasks/pluginmachinery/tasklog/template.go

```diff
@@ -0,0 +1,127 @@
+package tasklog
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/lyft/flyteidl/gen/pb-go/flyteidl/core"
+)
+
+// A simple log plugin that supports templates in urls to build the final log link. Supported templates are:
+// {{ .podName }}: Gets the pod name as it shows in k8s dashboard,
+// {{ .namespace }}: K8s namespace where the pod runs,
+// {{ .containerName }}: The container name that generated the log,
+// {{ .containerId }}: The container id docker/crio generated at run time,
+// {{ .logName }}: A deployment specific name where to expect the logs to be.
+// {{ .hostname }}: The hostname where the pod is running and where logs reside.
+type TemplateLogPlugin struct {
+	templateUris  []string
+	messageFormat core.TaskLog_MessageFormat
+}
+
+type regexValPair struct {
+	regex *regexp.Regexp
+	val   string
+}
+
+var podNameRegex = mustCreateRegex("podName")
```

Comment:
do you want to put these in a struct and initialize the struct? That way if we find that this is becoming expensive we can add it to the initialization hook?

Reply:
I moved it to a struct and added a benchmark test for it... it's at 0.026914 millisecond at the moment... should be fine...